src/mtype.d

```diff
@@ -250,7 +250,8 @@ extern (C++) Type stripDefaultArgs(Type t)
                             for (size_t j = 0; j < params.dim; j++)
                                 (*params)[j] = (*parameters)[j];
                         }
-                        (*params)[i] = new Parameter(p.storageClass, ta, null, null);
+                        StorageClass stc = p.storageClass & (~STCauto); // issue 14656
+                        (*params)[i] = new Parameter(stc, ta, null, null);
```

Comment:
It had introduced by #4728. And later it's changed again back to the original code by #4729. On the other hand, current stable branch contains only #4728, so this change is actually reverting part of #4729.

#5032 is not directly related.